### PR TITLE
allow targetarch to be set by Docker buildkit

### PR DESCRIPTION
**Pull request by jakecoffman**
According to the [Docker docs](https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope), we shouldn't have to set the TARGETARCH if we simply declare it in the Dockerfiles as `ARG TARGETARCH` when using Docker buildkit. 

This simplifies things a bit and allows any other architectures that our scripts don't explicitly handle to build the containers.

I tested this on my ARM machine and it works.

### CODE REVIEW

File: Dockerfile.updater-core [hunk 1]
```diff
@@ -1,7 +1,5 @@
 FROM ubuntu:20.04 as builder
 
-ARG TARGETARCH=amd64
```
Comment (jakecoffman) [inline Dockerfile.updater-core:3]: This wasn't used in this file so I removed it.

**Comment by Nishnha** [comment]
Could this close out https://github.com/dependabot/dependabot-core/pull/5536?
That PR also adds a build matrix for the new target arch but I don't think that's necessary since we have codespaces

**Comment by jakecoffman** [comment]
@Nishnha no we still will need to add a workflow to build and publish for ARM to do that. This is just to clean up the scripts a bit. 

Also I had reverted some changes @jeffwidman had done in https://github.com/dependabot/dependabot-core/pull/6502 and while researching it a bit more realized we don't have to set that value at all.

**Review by jeffwidman — APPROVED** [review]
Oh nice, this is even cleaner than what I'd done!

**Pull request merged**